src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,168 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> addedNode = new Node<>(null, value, null);
+        if (head == null) {
+            head = tail = addedNode;
+        } else {
+            addLastNode(addedNode);
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndex(index, size + 1);
+        Node<T> addedNode = new Node<>(null, value, null);
+        if (head == null) {
+            head = tail = addedNode;
+        } else if (index == 0) {
+            addFirstNode(addedNode);
+        } else if (index == size) {
+            addLastNode(addedNode);
+        } else {
+            addMiddleNode(index, addedNode);
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index, size);
+        return getCurrentNode(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index, size);
+        Node<T> currentNode = getCurrentNode(index);
+        T result = currentNode.value;
+        currentNode.value = value;
+        return result;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index, size);
+        T result = get(index);
+        unlink(getCurrentNode(index));
```

Comment:
why do you need both `get` and `getCurrentNode` methods?

Reply:
It looks like you just call the method getCurrentNode twice (first in the method get() and second on line 65)